File: app/models/station.rb
class Station < ApplicationRecord
  belongs_to :trade_hub
  validates :trade_hub_id, :name, :cpp_station_id, presence: true

  def system
    trade_hub ? system.trade_hub : nil
  end

  def self.get_by_cpp_station_id(cpp_station_id)
    Station.where( 'cpp_station_id=?', cpp_station_id).first
  end

  def self.to_trade_hub_id(cpp_station_id)
    station = get_by_cpp_station_id(cpp_station_id)
    station && station.trade_hub ? station.trade_hub.id : nil
  end
end

File: evebs/models/tables/station.py
from evebs.extensions import db


class Station(db.Model):
    __tablename__ = 'stations'

    id = db.Column(db.Integer, primary_key=True)
    trade_hub_id = db.Column(db.Integer, db.ForeignKey('trade_hubs.id'))
    name = db.Column(db.String)
    cpp_station_id = db.Column(db.Integer)
    created_at = db.Column(db.DateTime)
    updated_at = db.Column(db.DateTime)

    trade_hub = db.relationship('TradeHub', back_populates='stations')

    @classmethod
    def to_trade_hub_id(cls, location_id):
        station = cls.query.filter_by(cpp_station_id=location_id).first()
        return station.trade_hub_id if station else None
